iguana.corecontroller/src/main/java/org/aksw/iguana/cc/utils/FileUtils.java

```diff
@@ -14,74 +14,78 @@
 	 */
 	public static int countLines(File filename) throws IOException {
-		try (InputStream is = new BufferedInputStream(new FileInputStream(filename))) {
-
-			byte[] c = new byte[1024];
-			int count = 0;
-			int readChars;
-			boolean empty = true;
-			byte lastChar = '\n';
-			while ((readChars = is.read(c)) != -1) {
-				for (int i = 0; i < readChars; ++i) {
-					if (c[i] == '\n') {
-						// Check if line was empty
-						if (lastChar != '\n') {
-							++count;
+		if(getLineEnding((filename.getAbsolutePath())).equals("\n")) {
+			final int BUFFER_SIZE = 8192;
+			try (InputStream is = new BufferedInputStream(new FileInputStream(filename), BUFFER_SIZE)) {
+				byte[] c = new byte[BUFFER_SIZE];
+				int count = 0;
+				int readChars = 0;
+				boolean empty = true;
+				byte lastChar = '\n';
+				while ((readChars = is.read(c)) != -1) {
+					for (int i = 0; i < readChars; ++i) {
+						if (c[i] == '\n') {
+							// Check if line was empty
+							if (lastChar != '\n') {
+								++count;
+							}
+						} else {
+							empty = false;
 						}
-					} else {
-						empty = false;
+						lastChar = c[i];
 					}
-					lastChar = c[i];
 				}
+				if (lastChar != '\n') {
+					count++;
+				}
+				return (count == 0 && !empty) ? 1 : count;
 			}
-			if (lastChar != '\n') {
-				count++;
+		} else {
+			String line = "";
+			int count = 0;
+			try(BufferedReader br = new BufferedReader(new FileReader(filename))) {
+				while ((line = br.readLine()) != null) {
+					if (!line.isEmpty()) {
+						count++;
+					}
+				}
 			}
-			return (count == 0 && !empty) ? 1 : count;
+			return count;
 		}
 	}
 
 	/**
-	 * Returns a line at a given position of a File
-	 * 
-	 * @param pos      line which should be returned
-	 * @param filename File in which the queries are stated
-	 * @return line at pos
+	 * Returns a line at a given position of a File. <br/>
+	 * This method ignores every empty line, therefore the parameter <code>pos</code> references the n-th non-empty line.
+	 *
+	 * @param index the position of a non-empty line which should be returned
+	 * @param file 	the file to read from
+	 * @return the line at the given position
 	 * @throws IOException
 	 */
-	public static String readLineAt(int pos, File filename) throws IOException {
-		try (InputStream is = new BufferedInputStream(new FileInputStream(filename))) {
-			StringBuilder line = new StringBuilder();
+	public static String readLineAt(int index, File file) throws IOException {
+		String line = "";
+		int count = 0;
 
-			byte[] c = new byte[1024];
-			int count = 0;
-			int readChars;
-			byte lastChar = '\n';
-			while ((readChars = is.read(c)) != -1) {
-				for (int i = 0; i < readChars; ++i) {
-					if (c[i] == '\n') {
-						// Check if line was empty
-						if (lastChar != '\n') {
-							++count;
-						}
-					} else if (count == pos) {
-						// Now the line
-						line.append((char) c[i]);
+		try(BufferedReader br = new BufferedReader(new FileReader(file))) {
```

Comment:
Both must be managed by the try-with block. I also did that wrong in my code snippets. Otherwise, FileReader might not be closed on exceptions:
```
try (FileReader fr = new FileReader(path);
	         BufferedReader br = new BufferedReader(fr)) {
```

Please check this for all changes made in this PR and make an issue to check it throughout the code.

Reply:
Similar things only happen in the CLIWorker class on line 54 and ResultSizeRetriever class on line 22, if I'm not overlooking something. The ResultSizeRetriever class is actually not being used, therefore I will only create an issue for the CLIWorker